scipy/sparse/dia.py

```diff
@@ -323,10 +323,16 @@ def diagonal(self, k=0):
         if k <= -rows or k >= cols:
             return np.empty(0, dtype=self.data.dtype)
         idx, = np.nonzero(self.offsets == k)
-        first_col, last_col = max(0, k), min(rows + k, cols)
+        first_col = max(0, k)
+        last_col = min(rows + k, cols)
```

Comment:
Not really necessary to change.

Reply:
Ok no problem. Was a totally optional remark on my side :) I am merging then, thanks!